test/copy.bats

```diff
@@ -0,0 +1,40 @@
+load helpers
+
+function setup() {
+    stacker_setup
+}
+
+function teardown() {
+    cleanup
+}
+
+@test "built type layers are restored correctly" {
+    cat > stacker.yaml <<EOF
+parent:
+    from:
+        type: oci
+        url: $CENTOS_OCI
+    run: |
+        touch /root/parent
+        cat /proc/self/mountinfo
+child:
+    from:
+        type: built
+        tag: parent
+    copy:
+      - stacker://parent/root/parent -> /root/parent
```

Comment:
I thought I saw @mikemccracken complain elsewhere about the ascii art here.
I'm not fan.  We have a reasonable data format, why not use it?

```yaml
copy:
 - { src: "stacker://parent/root/parent", target: "/root/parent", perms: 0755, owner: "1001" }
 - ...
```
The choice to use ascii art bit us https://github.com/project-stacker/stacker/issues/41 also.

Reply:
makes the syntax a little verbose, but unambiguous

Reply:
one thought I had about this is that now people have to remember which of `src`, `source`, `dst`, `dest`, `destination`, `tgt`, `target`, etc are the right keys.

My suggestion is to just allow all of them.

Reply:
Generally speaking I have convinced myself that "Be liberal in what you accept, and conservative in what you send" is wrong.  It just gets you into a support issue.  Its confusing for users too, they see some examples that say 'src' and others that say 'source' and they don't know which is "right" (which even *is* right?). And you can't fix the situation without breaking a user.

The best thing to do is be consistent and give good errors:  unknown key 'src' in copy element 1.  Did you mean 'source' ?


That said, I do think allowing a string "0755" is helpful, compared to only allowing that to be an integer.  That said, the following actually is correct. yaml reads 0755 as base8.

```
$ echo 'myperms: 0755' | python3 -c 'import sys, yaml; print(yaml.safe_load(sys.stdin)["myperms"])'
493
```

Reply:
I realize... its funny.  *yaml* is being very liberal in what it accepts , and thats a win for us ;). Yaml being liberal though and "friendly" to its users also ends up confusing them lots of times, which I'm sure we've all ran into.

Reply:
OK, yeah that's sane. Good errors solves that problem in a better way.